### PR TITLE
Migrated to twig (Fork CMS 4.x)

### PR DESCRIPTION
**Pull request by wouterverstuyf**
Hello @ikoene 

Ik heb de slideshow module gemigreerd naar Twig, zodat deze compatibel is met de laatste versie van Fork CMS.

Test gerust even uit en laat eventuele feedback maar horen!

Thx.

Groeten
Wouter
www.webflow.be

### CODE REVIEW

**Comment by WouterSchoofs** [comment]
+1

File: src/Frontend/Modules/Slideshow/Layout/Templates/Category.html.twig [hunk 1]
```diff
@@ -15,7 +15,7 @@
       <ul>
         <li>
         {# Meta #}
-        {{ galleri.publish_on|spoon_date(dateFormatLong,LANGUAGE ) }}
```
Comment (ikoene) [inline src/Frontend/Modules/Slideshow/Layout/Templates/Category.html.twig]: Typo (`gallery`)

File: src/Frontend/Modules/Slideshow/Layout/Templates/Detail.html.twig [hunk 2]
```diff
@@ -25,11 +25,11 @@
           <div class="flexsliderwrap{{ gallery.id }}">
             <ul class="slides">
               {% for _itr_0 in slideshow %}
-              <li data-thumb="/src/Frontend/Files/_itr_0/thumbnails/{{ _itr_0.filename }}">
+              <li data-thumb="/src/Frontend/Files/slideshow/thumbnails/{{ _itr_0.filename }}">
```
Comment (ikoene) [inline src/Frontend/Modules/Slideshow/Layout/Templates/Detail.html.twig]: What's up with this var naming? `_itr_0.filename` 

Shouldn't it be something like `{{ slideshow.filename }}`

File: src/Frontend/Modules/Slideshow/Layout/Templates/Detail.html.twig [hunk 3]
```diff
@@ -25,11 +25,11 @@
...
                 {% if _itr_0.data.link %}
                 <a href="{{ _itr_0.data.link.url }}"{% if _itr_0.data.link.external %} target="_blank" rel="nofollow"{% endif %} title="{{ _itr_0.title }}">
                 {% endif %}
-                  <img src="/src/Frontend/Files/_itr_0/{{ _itr_0.filename }}" alt="{{ _itr_0.title }}" />
+                  <img src="/src/Frontend/Files/slideshow/{{ _itr_0.filename }}" alt="{{ _itr_0.title }}" />
```
Comment (ikoene) [inline src/Frontend/Modules/Slideshow/Layout/Templates/Detail.html.twig]: Same here.

**Comment by ikoene** [comment]
The category name is not parsed in the **confirmation message** when deleting a category.

schermafbeelding 2016-08-23 om 22 20 08

**Comment by ikoene** [comment]
Small styling issue on EditImage action.

schermafbeelding 2016-08-23 om 22 30 18

**Comment by ikoene** [comment]
When clicking on the **amount of images** to on the slideshow overview, you should be redirected to the images tab on the slideshow detail, which isn't the case. Probably has something to do with the url.

schermafbeelding 2016-08-23 om 22 32 55

schermafbeelding 2016-08-23 om 22 33 28

**Comment by ikoene** [comment]
The internal/external link logic from the old build is not present here. The **external link textfield** should only be visible when the **external link checkbox** is checked.

Furthermore, the external link textfield should only be validated when the external link checkbox is checked.

I've included a gif which demonstrates the correct behaviour.

schermafbeelding 2016-08-23 om 22 41 35

5c29fd7d58cde9ae8be10970aac03614

**Comment by ikoene** [comment]
The **Settings** tab should not be visible when you've opted *not* to have **settings per slideshow**.

schermafbeelding 2016-08-23 om 22 56 53


schermafbeelding 2016-08-23 om 22 54 05

**Comment by ikoene** [comment]
There are several styling issues with a rendered slideshow in the frontend. See screenshot for more details.

schermafbeelding 2016-08-23 om 23 02 24

**Comment by ikoene** [comment]
You should *not* be able to deleted a category that's used for a slideshow. 

Currently, you can do this. Not only is the category deleted, but the slideshows using the category are not visible in the overview.

schermafbeelding 2016-08-23 om 23 07 26

File: src/Backend/Modules/Slideshow/Layout/Templates/Add.html.twig [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+              </div>
+              <div class="col-md-4">
+                <div class="panel panel-default">
+                  <div class="panel-heading">
+                    <p>{{ 'lbl.Status'|trans|capitalize }}</p>
+                  </div>
+                  <div class="panel-body">
+                    <div class="form-group">
+                      <ul class="list-unstyled">
+                        {% for _itr_4 in hidden %}
```
Comment (ikoene) [inline src/Backend/Modules/Slideshow/Layout/Templates/Add.html.twig]: Not sure what this `_itr_4` naming thing is, but you should replace it by something more meaningful.

File: src/Backend/Modules/Slideshow/Layout/Templates/Edit.html.twig [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+              </div>
+              <div class="col-md-4">
+                <div class="panel panel-default">
+                  <div class="panel-heading">
+                    <p>{{ 'lbl.Status'|trans|capitalize }}</p>
+                  </div>
+                  <div class="panel-body">
+                    <div class="form-group">
+                      <ul class="list-unstyled">
+                        {% for _itr_4 in hidden %}
```
Comment (ikoene) [inline src/Backend/Modules/Slideshow/Layout/Templates/Edit.html.twig]: Same here.

File: src/Backend/Modules/Slideshow/Layout/Templates/EditImage.html.twig [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+              </div>
+              <div class="col-md-4">
+                <div class="panel panel-default">
+                  <div class="panel-heading">
+                    <p>{{ 'lbl.Status'|trans|capitalize }}</p>
+                  </div>
+                  <div class="panel-body">
+                    <div class="form-group">
+                      <ul class="list-unstyled">
+                        {% for _itr_4 in hidden %}
```
Comment (ikoene) [inline src/Backend/Modules/Slideshow/Layout/Templates/EditImage.html.twig]: Same here.

**Comment by ikoene** [comment]
A slideshow title is required, but the title label is missing an asterisk (*).

schermafbeelding 2016-08-23 om 23 19 10

**Comment by ikoene** [comment]
All in all good effort, but it needs some polish 😉

**Comment by wouterverstuyf** [comment]
@ikoene Thx, it's the polish that makes something really good ;)
I will take a look at it the following weeks, at the moment a bit busy, if someone wants to help you're welcome!

**Comment by ikoene** [comment]
Several `.DS_Store` files commited.

**Comment by ikoene** [comment]
Allright, making good progress 👍 

I don't know if it's just me, but the slideshow animation seems to be absent. It doesn't seem to **slide** or **fade**. I noticed that the animation is present twice (`animation` & `animation_type`) in the JS. Could that be the cause?

schermafbeelding 2016-08-27 om 23 36 34

**Comment by ikoene** [comment]
Also, there's something wrong with the `max-width` for a slideshow (maybe it's related to the previous issue). The max-width is not taken into account.

schermafbeelding 2016-08-27 om 23 48 11

**Comment by ikoene** [comment]
The code in the `Add` action (src/Backend/Modules/Slideshow/Actions/Add.php) seems to be entirely new if I check the diff (see **Files changed** tab). I'm not sure how this is possible since there are only small tweaks to the file...

File: src/Backend/Modules/Slideshow/Layout/Templates/Add.html.twig [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+                    </div>
+                  </div>
+                </div>
+                <div class="panel panel-default">
+                  <div class="panel-heading">
+                    <p>{{ 'lbl.Dimensions'|trans|capitalize }}</p>
+                  </div>
+                  <div class="panel-body">
+                    <div class="form-group">
+                        <label for="height" class="control-label">{{ 'lbl.Width'|trans|capitalize }}<abbr title="{{ 'lbl.RequiredField'|trans }}">*</abbr></label>
```
Comment (ikoene) [inline src/Backend/Modules/Slideshow/Layout/Templates/Add.html.twig]: Should be `for="width"`.

File: src/Frontend/Modules/Slideshow/Layout/Widgets/Slideshow.html.twig [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+#}
+
+  {% if widgetGallery %}
+    {# Title #}
+      <h3>
+        {{ widgetGallery.title }}
+      </h3>
+
+    {# Slideshow #}
+      <div class="flex-container" id="flex{{ widgetGallery.id }}" style="max-width:widgetGallery.widthpx">
```
Comment (ikoene) [inline src/Frontend/Modules/Slideshow/Layout/Widgets/Slideshow.html.twig]: Brackets are missing around `widgetGallery.width`.

File: src/Backend/Modules/Slideshow/Layout/Templates/Add.html.twig [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+{% block actionbar %}
+
+{% endblock %}
+
+{% block content %}
+
+  {% form add %}
+    {% if categories %}
+
+  <div class="row fork-module-content">
```
Comment (ikoene) [inline src/Backend/Modules/Slideshow/Layout/Templates/Add.html.twig]: From here on the indentation is not correct.

File: src/Backend/Modules/Slideshow/Layout/Templates/Edit.html.twig [hunk 10]
```diff
@@ -0,0 +1,338 @@
...
+                    </div>
+                  </div>
+                </div>
+                <div class="panel panel-default">
+                  <div class="panel-heading">
+                    <p>{{ 'lbl.Dimensions'|trans|capitalize }}</p>
+                  </div>
+                  <div class="panel-body">
+                    <div class="form-group">
+                        <label for="height" class="control-label">{{ 'lbl.Width'|trans|capitalize }}<abbr title="{{ 'lbl.RequiredField'|trans }}">*</abbr></label>
```
Comment (ikoene) [inline src/Backend/Modules/Slideshow/Layout/Templates/Edit.html.twig]: Should be `for="width"`.

File: src/Backend/Modules/Slideshow/Layout/Templates/EditCategory.html.twig [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+            </button>
+          {% endif %}
+        </div>
+        <div class="btn-group pull-right" role="group">
+          <button id="editButton" type="submit" name="edit" class="btn btn-primary">
+            <span class="fa fa-check"></span>{{ 'lbl.Publish'|trans|capitalize }}
+          </button>
+        </div>
+      </div>
+      {% if showSlideshowDeleteCategory %}
```
Comment (ikoene) [inline src/Backend/Modules/Slideshow/Layout/Templates/EditCategory.html.twig]: This should probably be `{% if deleteCategoryAllowed %}`.

File: src/Backend/Modules/Slideshow/Layout/Templates/EditImage.html.twig [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+          <span class="modal-title h4">{{ 'lbl.Delete'|trans|capitalize }}</span>
+        </div>
+        <div class="modal-body">
+          <p>{{ 'msg.ConfirmDelete'|trans|sprintf(item.title) }}</p>
+        </div>
+        <div class="modal-footer">
+          <button type="button" class="btn btn-default" data-dismiss="modal">
+            <span class="fa fa-times"></span>{{ 'lbl.Cancel'|trans|capitalize }}
+          </button>
+          <a href="{{ geturl('DeleteImage' ) }}&amp;id={{ item.id }}&amp;gallery_id={{ item.gallery_id }}" class="btn btn-danger">
```
Comment (ikoene) [inline src/Backend/Modules/Slideshow/Layout/Templates/EditImage.html.twig]: Superflous space after 'DeleteImage'.

`{{ geturl('DeleteImage') }}`

File: src/Backend/Modules/Slideshow/Layout/Templates/Edit.html.twig [hunk 13]
```diff
@@ -0,0 +1,338 @@
...
+                {% if not dataGrid %}
+                  {{ 'msg.NoItems'|trans }}
+                {% endif %}
+              </div>
+            </div>
+
+            {% if showSlideshowAddImage %}
+            <div class="row">
+              <div class="col-md-12">
+                <a href="{{ geturl('add_image' ) }}&amp;id={{ item.id }}" class="btn btn-primary">
```
Comment (ikoene) [inline src/Backend/Modules/Slideshow/Layout/Templates/Edit.html.twig]: Superfluous space after `add_image`.

`geturl('add_image')`

File: src/Frontend/Modules/Slideshow/Layout/Templates/Category.html.twig [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+
+    {# Title #}
+      <h3>
+        <a href="{{ full_url }}/{{ gallery.meta_url }}">{{ gallery.title }}</a>
+      </h3>
+
+      <ul>
+        <li>
+        {# Meta #}
+        {{ gallery.publish_on|spoondate(dateFormatLong,LANGUAGE ) }}
```
Comment (ikoene) [inline src/Frontend/Modules/Slideshow/Layout/Templates/Category.html.twig]: Superfluous space after `LANGUAGE`.

`spoondate(dateFormatLong,LANGUAGE)`

**Review by ikoene — CHANGES_REQUESTED** [review]
Hey @wouterverstuyf Checking out this new review feature in GitHub to check on your progress :) How's it going? Need any help?

**Comment by wouterverstuyf** [comment]
Hi @ikoene, It's still on the to do list, but I'm full with some projects at the moment. I try to find time soon... Of course, any help can speed up the progress :)

**Comment by ikoene** [comment]
Good job 👍 Did some testing, works great!

**Review by ikoene — APPROVED** [review]

**Pull request merged**